spec/lita/services/market_manager_spec.rb

```diff
@@ -221,90 +236,164 @@ def setup_lunchers
+
+      before do
+        add_limit_order(oscar, 'bid', first_tx_time + 2.days, 1)
+        add_limit_order(fernanda, 'bid', first_tx_time + 3.days, 1)
+        add_limit_order(andres, 'ask', first_tx_time, ask_price)
+        add_limit_order(fdom, 'bid', first_tx_time + 1.day, bid_price)
+        setup_lunchers
+      end
+
+      context 'with ask price greater than bid price' do
+        let(:ask_price) { 3 }
+        let(:bid_price) { 2 }
+
+        it 'does not execute a tx' do
+          size_before_execution = subject.ask_orders.size
+          subject.execute_transaction
+          expect(subject.ask_orders.size).to eq(size_before_execution)
+        end
+      end
+
+      context 'with bid price grater than ask price' do
+        let(:ask_price) { 2 }
+        let(:bid_price) { 3 }
+
+        it 'executes a tx at ask price' do
+          tx = subject.execute_transaction
+          expect(subject.ask_orders.size).to eq(0)
+          expect(tx['price']).to eq(ask_price)
+        end
+      end
+
+      context 'with bid price equal to ask price' do
+        let(:ask_price) { 3 }
+        let(:bid_price) { 3 }
+
+        it 'executes the tx at ask price' do
+          tx = subject.execute_transaction
+          expect(subject.ask_orders.size).to eq(0)
+          expect(tx['price']).to eq(ask_price)
+        end
+      end
+
+      context 'with two equal ask orders but added at different time' do
+        let(:ask_price) { 3 }
+        let(:bid_price) { 3 }
+        let(:other_user_1) { Lita::User.create(201, mention_name: 'pepito') }
+        let(:other_user_2) { Lita::User.create(202, mention_name: 'pepito2') }
+        let!(:other_orders) do
+          [
+            add_limit_order(other_user_1, 'ask', first_tx_time - 1.day, ask_price),
+            add_limit_order(other_user_2, 'ask', first_tx_time + 1.day, ask_price)
+          ]
+        end
+        let(:older_order) { other_orders.first }
+
+        it 'executes the tx at ask price and with the oldest order' do
+          original_ask_orders_size = subject.ask_orders.size
+          tx = subject.execute_transaction
+          expect(subject.ask_orders.size).to eq(original_ask_orders_size - 1)
+          expect(tx['price']).to eq(ask_price)
+          expect(tx['ask_order']['id']).to eq(older_order[:id])
+        end
+
+        it 'second execution matches the second oldest order' do
+          subject.execute_transaction
+          add_limit_order(other_user_1, 'bid', first_tx_time + 4.days, ask_price)
+          tx = subject.execute_transaction
+          expect(tx['ask_order']['user_id']).to eq(andres.id)
+        end
       end
     end
   end
 
   describe '#transaction_possible?' do
     context 'transaction possible' do
       it 'returns true' do
-        add_limit_order(SecureRandom.uuid, oscar, 'ask', Time.new(2018, 10, 5))
-        add_limit_order(SecureRandom.uuid, fdom, 'bid', Time.new(2018, 10, 3))
+        add_limit_order(oscar, 'ask', Time.new(2018, 10, 5))
```

Comment:
Do not use Time.new without zone. Use one of Time.zone.local, Time.current, Time.new.in_time_zone, Time.new.utc, Time.new.getlocal, Time.new.xmlschema, Time.new.iso8601, Time.new.jisx0301, Time.new.rfc3339, Time.new.httpdate, Time.new.to_i, Time.new.to_f instead.